### ISSUE TITLE
False positive for CA2000 in a specific scenario

### ISSUE THREAD

**Issue by nick-morhun** [dotnet/roslyn-analyzers#7648]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 9.0.0

### Describe the bug

False positive detection for the following code (or when using the "using" keyword):

```
using System;
using System.Net.Http;
using System.Threading.Tasks;

class TestClass1 : IDisposable
{
  public void Dispose()
  {
    throw new NotImplementedException();
  }
}

class TestClass
{
  public void Test()
  {
    using var _ = new TestClass1();

    for (; true; )
    {
      // Next line is reported as an issue
      var test = new TestClass1();

      try
      {
        Test1();
      }
      finally
      {
        test.Dispose();
      }
    }
  }

  private async Task Test1()
  {
    using var _ = new StreamContent(null);
  }
}
```

### Expected behavior

No issues in the marked line.

### Actual behavior

"CA2000: Use recommended dispose pattern to ensure that object created by 'new TestClass1()' is disposed on all paths. If possible, wrap the creation within a 'using' statement or a 'using' declaration. Otherwise, use a try-finally pattern, with a dedicated local variable declared before the try region and an unconditional Dispose invocation on non-null value in the 'finally' region, say 'x?.Dispose()'. If the object is explicitly disposed within the try region or the dispose ownership is transfered to another object or method, assign 'null' to the local variable just after such an operation to prevent double dispose in 'finally'."
